packages/block-library/src/gallery/shared.js

```diff
@@ -21,3 +21,11 @@ export const pickRelevantMediaFiles = ( image, sizeSlug = 'large' ) => {
 	}
 	return imageProps;
 };
+
+export function isGalleryV2Enabled() {
+	if ( typeof process !== 'undefined' && process.env?.NODE_ENV === 'test' ) {
+		return true;
+	}
+
+	return window.wp.galleryBlockV2Enabled;
```

Comment:
Tiniest of tiny nits, but since this function is called `isGalleryV2Enabled`, should this return a boolean (e.g. `!!`)? Also, I assume that the `wp` object will always be available in practice, but I see that elsewhere in the Gallery block where `window.wp` is accessed, we've used optional chaining. Not sure if it's needed here, too?

Reply:
> should this return a boolean

Hm.. `window.wp.galleryBlockV2Enabled` already is a boolean, I wonder, do you mean to coerce with `!!` for the case where it is `undefined`?

> Not sure if it's needed here, too?

Good question.. I had contemplated that as well.. my thinking at the time (at least from the mobile perspective) is that `window.wp` should be defined (and as an `Object`), and that since we are altering user content on the basis of this flag, it'd be better to fail fast if that assumption is incorrect, to avoid incorrectly transforming the content. Otoh, this is function is shared cross-platform, so it'd be good to know from the web side whether this assumption is valid - and whether or not failing fast is appropriate there as well in case it's `undefined`. :thinking: 

Also, maybe it's actually fine on mobile and web to default to false(y) when the assumption is not met? :thinking: One of the hesitations I originally had was born out of the race conditions we've encountered while testing / debugging these flows - not sure whether the presence (or absence) of `window.wp`, if even possible, could also be a timing-based condition?

Reply:
@mkevins, @andrewserong I wonder if we want to default to v2 being enabled if `window.wp.galleryBlockV2Enabled` is not explicitly set to `false`, eg. `return window.wp?.galleryBlockV2Enabled === false ? false : true;`  - the reason being that the disabled case with `use_balanceTags === 1` and `WP < 5.9` is the edge case so are we better to assume we use v2 if `wp.galleryBlockV2Enabled` was undefined for some reason?

Reply:
Thanks for the thoughts Matthew and Glen!

>  I wonder, do you mean to coerce with `!!` for the case where it is `undefined`?

Yes, I was mostly thinking of how do we want to handle the case of `undefined`.

> I wonder if we want to default to v2 being enabled if `window.wp.galleryBlockV2Enabled` is not explicitly set to false

That logic sounds good to me, as you say, the presence of the flag is there to deal with an explicit edge case, so handling it with an explicit check for `false` sounds good to me 👍